weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/TraceUsageStats.tsx

```diff
@@ -152,89 +116,39 @@ export const getTokensAndCostFromUsage = (usage: {
   const tokensNum = metrics.inputs.tokens.total + metrics.outputs.tokens.total;
   const tokens = formatTokenCount(tokensNum);
 
-  const tooltipRowStyles = {
-    display: 'flex',
-    justifyContent: 'space-between',
-    alignContent: 'center',
-    gap: '16px',
-  };
-
-  const tooltipDivider = (
-    <Divider
-      sx={{
-        borderColor: MOON_600,
-        marginTop: '8px',
-        marginBottom: '7px',
-      }}
-    />
-  );
-
-  const tokenToolTip = (
-    <Box>
-      {Object.keys(metrics.inputs.tokens).map(model => (
-        <Box key={model + 'input'} sx={tooltipRowStyles}>
-          <span>{model === 'total' ? 'Input tokens' : model}: </span>
-          <span>
-            {FORMAT_NUMBER_NO_DECIMALS.format(metrics.inputs.tokens[model])}
-          </span>
-        </Box>
-      ))}
-      {tooltipDivider}
-      {Object.keys(metrics.outputs.tokens).map(model => (
-        <Box key={model + 'output'} sx={tooltipRowStyles}>
-          <span>{model === 'total' ? 'Output tokens' : model}: </span>
-          <span>
-            {FORMAT_NUMBER_NO_DECIMALS.format(metrics.outputs.tokens[model])}
-          </span>
-        </Box>
-      ))}
-    </Box>
-  );
+  const costToolTip = <CostToolTip {...metrics} />;
+  const tokenToolTip = <TokenToolTip {...metrics} />;
 
-  const costToolTip = (
-    <Box>
-      <span style={{fontWeight: 600}}>Estimated Cost</span>
-      {Object.keys(metrics.inputs.cost).map(model => (
-        <Box key={model + 'input'} sx={tooltipRowStyles}>
-          <span>{model === 'total' ? 'Input cost' : model}: </span>
-          <span>{formatTokenCost(metrics.inputs.cost[model])}</span>
-        </Box>
-      ))}
-      {tooltipDivider}
-      {Object.keys(metrics.outputs.cost).map(model => (
-        <Box key={model + 'output'} sx={tooltipRowStyles}>
-          <span>{model === 'total' ? 'Output cost' : model}: </span>
-          <span>{formatTokenCost(metrics.outputs.cost[model])}</span>
-        </Box>
-      ))}
-    </Box>
-  );
   return {costNum, cost, tokensNum, tokens, costToolTip, tokenToolTip};
 };
 
-const TraceStat = ({
-  icon,
-  label,
-  tooltip,
-}: {
-  icon?: IconName;
-  label: string;
-  tooltip?: ReactNode;
-}) => {
-  const trigger = (
-    <div>
-      <Pill
-        icon={icon}
-        label={label}
-        color="moon"
-        className="bg-transparent text-moon-500 dark:bg-transparent dark:text-moon-500"
-      />
-    </div>
-  );
+export const getTokensFromCellParams = (params: {[key: string]: any}) => {
```

Comment:
everything except for this last function is pulling things out to costTypes and costUtils